highway_env/envs/racetrack_env.py

```diff
@@ -77,7 +80,12 @@ def _rewards(self, action: np.ndarray) -> dict[str, float]:
         }
 
     def _is_terminated(self) -> bool:
-        return self.vehicle.crashed
+        if self.config["terminate_off_road"]:
+            while self.vehicle.on_road:
```

Comment:
why the while loop here? just return self.vehicle.crashed or not self.vehicle_on_road